azurerm/data_source_arm_scheduled_time.go

```diff
@@ -0,0 +1,172 @@
+package azurerm
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+type Frequency string
+
+const (
+	Day     Frequency = "Day"
+	Hour    Frequency = "Hour"
+	Month   Frequency = "Month"
+	OneTime Frequency = "OneTime"
+	Week    Frequency = "Week"
+)
+
+func dataSourceArmScheduledTime() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmScheduledTimeRead,
+		Schema: map[string]*schema.Schema{
+			"frequency": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(Day),
+					string(Hour),
+					string(Month),
+					string(OneTime),
+					string(Week),
```

Comment:
it'd be good to add an acceptance test for each of these if possible

also: does `OneTime` make sense in this context (as a data source) or would you be more likely to know that and specify the time period? 🤔

Reply:
I added tests for all the possible periodicity. I can imagine a case when the `OneTime` make sense, so i would leave it there if you don't mind.